crates/sqlbuiltins/src/functions/table/flightsql.rs

```diff
@@ -0,0 +1,69 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use datafusion::arrow::datatypes::DataType;
+use datafusion::datasource::TableProvider;
+use datafusion::logical_expr::{Signature, Volatility};
+use datafusion_ext::errors::{ExtensionError, Result};
+use datafusion_ext::functions::{FuncParamValue, TableFuncContextProvider};
+use datasources::flightsql::{FlightSqlSourceConnectionOptions, FlightSqlSourceProvider};
+use protogen::metastore::types::catalog::{FunctionType, RuntimePreference};
+
+use super::TableFunc;
+use crate::functions::ConstBuiltinFunction;
+
+#[derive(Debug, Clone, Copy)]
+pub struct ReadFlightSql;
+
+impl ConstBuiltinFunction for ReadFlightSql {
+    const NAME: &'static str = "read_flightsql";
+    const DESCRIPTION: &'static str = "Reads a FlightSQL table";
+    const EXAMPLE: &'static str =
+        "SELECT * FROM read_flightsql('https://localhost:37019', 'database', 'token')";
```

Comment:
I don't think `database` and `token` should be specified. We should instead allow users to pass in headers as each flightsql implementation likely has it's own requirements.

```sql
read_flightsql('https://localhost:37019', authorization => '<token>', database => 'database', some_other_header => 'some_value')"
```

Reply:
to be honest, this was just a prototype; my inclination is that we'll actually make this function be `read_influxdb` and avoid providing generic flightsql cases, for exactly the reasons you have (though I think we can entertain the potential fantasy that flightsql is actually generic.)